### PR TITLE
[New Exercise] Micro blog

### PR DESCRIPTION
**Pull request by Cool-Katt**
This time I also added some approaches too.

### CODE REVIEW

**Comment by Cool-Katt** [comment]
Okay, I may have messed up the approaches and then messed up more trying to fix the mistake but the CI checks all pass now soI hope this can work. 

Sorry

**Comment by Cool-Katt** [comment]
Anything else needed to merge this? Let me know.

File: exercises/practice/micro-blog/.approaches/iterators/content.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+console.log(splitWithIterator(string)); // will be "👨‍👨‍👧" - incorrect
+console.log(splitWithIterator(string2)); // will be "‍The q"
+```
+
+This solution:
+
+- Uses [spread syntax][spread] to unpack the string into an array of its characters.
+  - internaly, the spread operator works with iterators to separate the string by its code points.
+- Then it separates the first 5 characters (code points).
+- Finally, it joins them back into a string.
```
Comment (SleeplessByte) [inline exercises/practice/micro-blog/.approaches/iterators/content.md:18]: Can you add why it yields the incorrect solution?

File: exercises/practice/micro-blog/.approaches/regex/content.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+console.log(splitWithRegEx(string)); // will be "👨‍👨‍👧" - incorrect
+console.log(splitWithIterator(string2)); // will be "‍The q"
+```
+
+This solution:
+
+- Uses the [String.match() method][match] with a supplied RegEx
+  - The RegEx supplied matches any character `.`, between 0 and 5 times `{0, 5}`. The `u` flag enables Unicode support.
+  - This matches characters by code points as well.
+- Then it returns the first match as the output string.
```
Comment (SleeplessByte) [inline exercises/practice/micro-blog/.approaches/regex/content.md:18]: Can you add why it yields the incorrect solution?

**Review by SleeplessByte — APPROVED** [review]
Yeah looking good

File: exercises/practice/micro-blog/.approaches/iterators/content.md (hunk 1 above)
Comment (Cool-Katt) [reply exercises/practice/micro-blog/.approaches/iterators/content.md:18]: done

File: exercises/practice/micro-blog/.approaches/regex/content.md (hunk 2 above)
Comment (Cool-Katt) [reply exercises/practice/micro-blog/.approaches/regex/content.md:18]: also done

**Pull request merged**